### PR TITLE
Rename `ScriptType` to `KeychainKind`

### PR DESCRIPTION
**Pull request by afilini**
### Description

This rename was suggested in #210 and it has since been added to #103. 

The `KeychainKind` name seems more appropriate for an enum that is used to determine whether something (like an address) is part of the internal or external descriptor (i.e. keychain). `ScriptType` could be easily confused for "the type of script", like a witness or non-witness script hash.

The `CHANGELOG.md` has also been updated to list this API-breaking change.

### Notes to the reviewers

The substitution was done using `sed`, so there could be some misspelled variable names hidden around the code that haven't been replaced. I couldn't find any by looking around manually but the diff is rather large, so there's a chance I've missed one or two. Either way, I don't think we should spend too much time hunting for those: if you can find one, please let me know, but if we miss a couple it won't be the end of the world.

For reference, the `sed` commands I used are:

```
find -type f -name *.rs -exec sed -i 's/ScriptType/KeychainKind/g' {} \;
find -type f -name *.rs -exec sed -i 's/script_type/keychain/g' {} \;
```

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### New Features:

* [x] I've updated `CHANGELOG.md`

#### Bugfixes:

* [x] This pull request breaks the existing API
* [x] I'm linking the issue being fixed by this PR

### CODE REVIEW

**Comment by LLFourn** [comment]
See https://github.com/afilini/bdk/pull/1 for more comment fixes. Everything reads much better after this. Thanks!

After thinking about it you could actually do  s/KeychainKind/Keychain/g if you wanted. Although appending `Kind` to these kind of enums is best practice I don't *think* we'll ever have a struct that is called just `Keychain` since it's an abstraction for a set of keys within a wallet.

**Comment by afilini** [comment]
I'd keep it this way because potentially in the future we could actually implement a "Keychain", a secure storage for keys.

I would like to see if we can for instance integrate with the OS' APIs on mobile platforms to encrypt secrets in the secure element, make them available with biometrics data and things like that..

File: src/database/mod.rs [hunk 1]
```diff
@@ -95,58 +95,58 @@ pub trait BatchOperations {
...
     fn get_raw_tx(&self, txid: &Txid) -> Result<Option<Transaction>, Error>;
     /// Fetch the transaction metadata and optionally also the raw transaction
     fn get_tx(&self, txid: &Txid, include_raw: bool) -> Result<Option<TransactionDetails>, Error>;
-    /// Return the last defivation index for a script type
-    fn get_last_index(&self, script_type: ScriptType) -> Result<Option<u32>, Error>;
+    /// Return the last defivation index for a keychain.
+    fn get_last_index(&self, keychain: KeychainKind) -> Result<Option<u32>, Error>;
 
-    /// Increment the last derivation index for a script type and returns it
+    /// Increment the last derivation index for a keychain and return it
```
Comment (RCasatta) [inline src/database/mod.rs:146]: wasn't returns with s correct?

**Comment by RCasatta** [comment]
apart from the nit utack 9b31ae9153e590d06835f5e49ffb643116334c26

File: src/database/mod.rs (hunk 1 above)
Comment (afilini) [reply src/database/mod.rs:146]: I guess this was done for consistency, since most of the other methods are not in the third person

Comment (LLFourn) [reply src/database/mod.rs:146]: it could be "increment**s**" with "return**s**" -- that might have been better but they have to match :)

**Pull request merged**